### PR TITLE
ARM disassembler: don't compute [pc, reg] memory location

### PR DESCRIPTION
**Pull request by pluswave**
- [x] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

ARM disassembler: for [pc, #imm ] , is ok to compute the address to display out. but not for [pc, reg].

### CODE REVIEW

File: libr/parse/p/parse_arm_pseudo.c [hunk 1]
```diff
@@ -360,16 +360,19 @@ static bool subvar(RParse *p, RAnalFunction *f, ut64 addr, int oplen, char *data
 			if (!ripend) {
 				ripend = "]";
 			}
-			if (neg) {
-				repl_num -= r_num_get (NULL, neg + 1);
-			} else {
-				repl_num += r_num_get (NULL, rip);
+			char * maybe_num = neg? neg+1 : rip;
+			if( r_is_valid_input_num_value(NULL, maybe_num)){
```
Comment (trufae) [inline libr/parse/p/parse_arm_pseudo.c:364]:
```suggestion
			if (r_is_valid_input_num_value (NULL, maybe_num)) {
```

File: libr/parse/p/parse_arm_pseudo.c [hunk 2]
```diff
@@ -360,16 +360,19 @@ static bool subvar(RParse *p, RAnalFunction *f, ut64 addr, int oplen, char *data
...
+				if (neg) {
+					repl_num -= r_num_get(NULL, maybe_num);
```
Comment (trufae) [inline libr/parse/p/parse_arm_pseudo.c:366]:
```suggestion
					repl_num -= r_num_get (NULL, maybe_num);
```

File: libr/parse/p/parse_arm_pseudo.c [hunk 3]
```diff
@@ -360,16 +360,19 @@ static bool subvar(RParse *p, RAnalFunction *f, ut64 addr, int oplen, char *data
...
+				} else {
+					repl_num += r_num_get(NULL, maybe_num);
```
Comment (trufae) [inline libr/parse/p/parse_arm_pseudo.c:368]:
```suggestion
					repl_num += r_num_get (NULL, maybe_num);
```

File: libr/anal/p/anal_arm_cs.c [hunk 4]
```diff
@@ -4083,7 +4083,7 @@ jmp $$ + 4 + ( [delta] * 2 )
 			op->stackop = R_ANAL_STACK_GET;
 			op->stackptr = 0;
 			op->ptr = -MEMDISP (1);
-		} else if (REGBASE(1) == ARM_REG_PC) {
+		} else if (REGBASE(1) == ARM_REG_PC && !HASMEMINDEX(1)) {
```
Comment (trufae) [inline libr/anal/p/anal_arm_cs.c:4086]:
```suggestion
		} else if (REGBASE (1) == ARM_REG_PC && !HASMEMINDEX (1)) {
```

**Review by trufae — APPROVED** [review]

**Comment by trufae** [comment]
Looks good! thank you!

**Comment by trufae** [comment]
do you have a test case to add? that may help us not to break things in the pending refactoring in rparse!

**Comment by radare** [comment]
Sorry i had to revert the patch because it was breaking a bunch of tests i missed during the review. Please feel free to resubmit the patch and add tests for it

Thanks

**Pull request merged**